### PR TITLE
`EventReader::last` to clear all events and return the last one

### PR DESCRIPTION
**Pull request by MiniaczQ**
# Objective

Add a way to retrieve only the last event from an `EventReader`.

## Solution

Implement `EventReader::last` which works much like `EventReader::clear`.

## Changelog

- Added `EventReader::last` to clear pending events and retrieve the last one.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Review by NiseVoid — APPROVED** [review]
I end up using `.read().last()` very often on events, especially for things like the window events (`WindowResized`, `WindowScaleFactorChanged`, etc) so this seems like a reasonable addition

File: crates/bevy_ecs/src/event.rs [hunk 1]
```diff
@@ -728,6 +735,17 @@ impl<E: Event> ManualEventReader<E> {
     pub fn clear(&mut self, events: &Events<E>) {
         self.last_event_count = events.event_count;
     }
+
+    /// See [`EventReader::last()`]
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/event.rs:739]:
```suggestion
    /// Consumes all available events and returns the last one.
    ///
    /// This works much like [`ManualEventReader::clear()`] except it returns the last event.
```

**Comment by alice-i-cecile** [comment]
Actually, quick sanity check: why is this *better* than `EventReader.read().last()`? Is it faster?

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Hmm. I don't like this on the basis of pure ergonomics. We should stick to the standard Rust API for things like this, rather than duplicating it.

What I would be in favor of is overwriting the default impl of `last` on the `Iterator` trait for `EventIterator` to get these benefits without adding more API surface 🤔 Even then though, this seems not particularly worth it and hard to ensure persists as our event backing changes or becomes more flexible.

**Comment by MiniaczQ** [comment]
Benchmarking shows `read().last()` to be 3x more performant than my `last()` implementation, which is extremely weird, but I'll take that.
Closed in favor of `read().last()`

**Pull request closed without merging**